### PR TITLE
1073 make external callbacks optional

### LINKED ISSUE #1073: Make running callbacks in a process optional
I was too trigger happy merging https://github.com/DiamondLightSource/hyperion/pull/1065. Given it causes us risk we should have gated it behind a property of some sort. Ideally it would be a GDA property so that the beamline scientist can revert it if there are issues. If we do sufficient beamline testing with this turned on we can remove the property.

## Acceptance Criteria
* There is a GDA property that we can use to switch callbacks running in the same process or not

### PR DESCRIPTION
**Pull request by dperl-dls**
Fixes #1073

Requires GDA change https://gerrit.diamond.ac.uk/c/gda/gda-mx/+/40900 to take effect, but shouldn't break anything on its own

### To test:
1. Run tests
2. Run `hyperion_restart()` from GDA (with the above change applied) and see that hyperion is started with the external callbacks turned off

### CODE REVIEW

File: run_hyperion.sh [hunk 1]
```diff
@@ -124,20 +128,37 @@ if [[ $START == 1 ]]; then
...
 
     #Add future arguments here
-    declare -A args=( ["IN_DEV"]="$IN_DEV" ["SKIP_STARTUP_CONNECTION"]="$SKIP_STARTUP_CONNECTION" ["VERBOSE_EVENT_LOGGING"]="$VERBOSE_EVENT_LOGGING"
-                    ["LOGGING_LEVEL"]="$LOGGING_LEVEL")
-    declare -A arg_strings=( ["IN_DEV"]="--dev" ["SKIP_STARTUP_CONNECTION"]="--skip-startup-connection" ["VERBOSE_EVENT_LOGGING"]="--verbose-event-logging"
-                            ["LOGGING_LEVEL"]="--logging-level=$LOGGING_LEVEL")
-
-    commands=()
-    for i in "${!args[@]}"
+    declare -A h_only_args=(        ["SKIP_STARTUP_CONNECTION"]="$SKIP_STARTUP_CONNECTION"
```
Comment (DominicOram) [inline run_hyperion.sh:131]: Could: Having just `h` here (and in the file as a whole) is not so obvious as to what you mean. Maybe `hyperion_main_only_args`?

File: src/hyperion/experiment_plans/experiment_registry.py [hunk 2]
```diff
@@ -54,8 +56,24 @@ def do_nothing():
...
+            GridscanInternalParameters
+            | GridScanWithEdgeDetectInternalParameters
+            | RotationInternalParameters
+            | PinCentreThenXrayCentreInternalParameters
+            | SteppedGridScanInternalParameters
+            | WaitForRobotLoadThenCentreInternalParameters
+        ]
+    )
+    experiment_param_type: type[AbstractExperimentParameterBase]
+    callback_collection_type: type[AbstractPlanCallbackCollection]
```
Comment (DominicOram) [inline src/hyperion/experiment_plans/experiment_registry.py:72]: ❤️

**Review by DominicOram — CHANGES_REQUESTED** [review]
Love all the additional typing, thanks! Could you add a couple of unit tests that cover the different behavior of `BlueskyRunner` depending on if the `use_external_callbacks` is set?

**Review by DominicOram — DISMISSED** [review]

**Review by olliesilvester — APPROVED** [review]
Looks good - that extra test seems trustworthy

**Pull request merged**